### PR TITLE
feat(console): explicit first-login plan choice replaces the trial prompt; pricing-page arrivals skip it

### PR DESCRIPTION
**Pull request by d-mo**
## Summary

The post-signup trial dialog is removed. In its place: a person who arrived from the pricing page is never asked anything, and a person who signed up from the plain signup page chooses a plan once, full screen, on their first console visit.

The dialog was confusing because it asked about a trial after signup, could be dismissed, and left the account on no plan in particular. The new screen asks the plain question ("which plan does this account run on") at the only moment it has to be answered, and cannot be dismissed.

## Founder decisions applied

1. **Pricing-page arrivals see no prompt of any kind.** Pro/Team/Business already go to Stripe first (the account is created from the completed session, and paying is the answer). Free now goes to `/register?plan=free` and straight to the console. The chosen plan is carried server side on the user row, stamped at registration from `plan_choice` on `POST /auth/register`, so it survives the email verification round trip, a link opened in another browser and a cleared cache. It is not localStorage and not only a query string.
2. **Plain signups get a full-screen plan choice on first login.** No console chrome, no dismiss, no "later". The same cards as `/console/settings/plan` and the public pricing page, from the same published content, with Free, Pro, Team and Business selectable and a quoted tier as contact. Free records the choice and reveals the console; a paid card opens Stripe with a card (trial terms as configured, stated on the card); a cancelled checkout returns to `/console`, which is this screen again, because nothing is recorded at the click. The choice is recorded on the user, so it never shows again, including on another device.
3. **Cloud only.** The endpoints live on the billing plugin, and the console has two gates in front of them: the `billing` feature must be on, and core's `/auth/users/me` must have said `plan_choice_made: false`. OSS makes no request and renders no screen. Existing users are covered by a backfill in the migration.

## The persisted field

`user.trial_prompt_dismissed_at` is **renamed** to `plan_choice_made_at` (migration `20260917_plan_choice`, chained off `20260917_onboarding_claim`). A rename rather than a new column, so every answer already recorded is preserved rather than re-asked. Null means the question is open, which is the only state the screen appears in. The timestamp is the whole record: which plan was picked is already written down as a subscription, or as its absence.

## The existing-user rule

A **backfill in the migration**, not a rule in the runtime check:

```sql
UPDATE "user" SET plan_choice_made_at = COALESCE(created_at, NOW())
WHERE plan_choice_made_at IS NULL
```

A "created before this shipped" clause in the check would need a hardcoded date living in the code forever, and would still have to be read on every request. The backfill settles it once, in the one place that knows what "before" means, and afterwards the check is a plain null test. `downgrade()` renames the column back and says in its docstring that the backfill is not undone.

Colleagues are covered separately, because they are created after the migration: `plugins/rbac` stamps `plan_choice_made_at` on admin user creation and on invitation acceptance. `not_billing_actor` in the plugin's check remains as defence in depth.

## Changes

Backend:
- `models/models/user.py`: column renamed, with a comment saying why it belongs on the user and not in the browser.
- `models/alembic/versions/20260917_plan_choice.py`: the rename plus the backfill.
- `schemas/auth.py`: `AuthUserCreate.plan_choice` (validated to a plan-id shape, dropped rather than refused when mangled, so a bad link can never cost a signup) and `AuthUserResponse.plan_choice_made` (defaults to `true`, so an older server never provokes the question).
- `api/auth/router.py`: `register()` stamps the row when a plan arrived; `/auth/users/me` and the register response report the flag.

Frontend:
- `components/plan-choice-screen.ts`: new, the full-screen choice. Cards from `loadPricingContent()` / `cloudPlans()`, quoted tiers detected by shape (both prices null) rather than by plan id.
- `components/trial-prompt.ts` and its test: deleted.
- `views/authed/console-shell.ts`: returns the screen from the top of `render()` for both `checking` and `required`, so the console is never painted and then pulled away. The router is untouched, so answering reveals the original destination, including deep links such as the CLI consent flow.
- `views/public/register-view.ts`: `parsePlanChoice()` (shape) and `isFreePlanId()` (does the id name a plan a signup can land on), plus the `plan_choice` field on the signup payload. A paid, quoted, invented or unverifiable id degrades to "ask on first login" rather than stamping a choice the account cannot have made. The price list is read only when a plan arrived on the link, so a plain signup makes no extra request.
- `views/public/pricing-view.ts`: an anonymous Free click now goes to `/register?plan=free`.
- `api.ts`: `getPlanChoice()` (passive, so a 402 or 429 cannot raise a dialog from a background read) and `recordFreePlanChoice()` replace the trial-prompt pair; `invalidateUserProfileCache()` added so answering does not have to blow away every cache in the app. A refusal that carries a reason is remembered for the tab, so the cohorts the plugin turns away without writing anything down (a member who cannot buy, an account already subscribed) ask once rather than on every route change; `unavailable` is deliberately not remembered, so an unreachable plugin still heals itself.

## Tests

| Suite | Result |
| --- | --- |
| `frontend/src/components/plan-choice-screen.test.ts` (new) | 11 passed |
| `frontend/src/views/authed/console-shell.test.ts` | 47 passed (40 before, 7 new) |
| `frontend/src/views/public/register-view.test.ts` | 35 passed (20 before, 15 new) |
| `frontend/src/views/public/pricing-view.test.ts` | 35 passed (33 before, 2 new) |
| `frontend/src/api.test.ts` | 67 passed |
| `npm test` (whole suite, 216 files) | 2981 passed, 0 failed |
| `npm run build` | clean |
| `npx tsc --noEmit` | clean |
| `backend/tests/test_plan_choice_signup.py` (new) | 9 passed |
| `backend/tests/models/test_alembic_single_head.py` | 5 passed |
| `backend/tests/{test_auth_schema,endpoints/test_auth,endpoints/test_auth_comprehensive,test_bootstrap_registration,test_require_email_verification}.py` | 143 passed with the above, 1 pre-existing failure (`test_failed_role_assignment_leaves_no_orphan`, fails identically on `main` at the same schema) |

What the new tests pin, by brief requirement:
- pricing-page arrival with a paid plan: no screen, goes to checkout (`pricing-view.test.ts`, "never routes a paid plan through signup");
- with Free: no screen, `/register?plan=free`, and the registration carries `plan_choice` (`pricing-view.test.ts`, `register-view.test.ts`);
- plain signup first login: the screen replaces the whole console, the choice is recorded, a second login shows nothing (`console-shell.test.ts`);
- OSS: no screen and zero requests, neither to `billing/plan-choice` nor to `/landing-content.json` (`console-shell.test.ts`, "does not exist without the billing plugin");
- the persisted choice, the plan surviving verification (verified through a client that shares nothing with the signup tab), and the existing-user rule (`test_plan_choice_signup.py`).

## Operator notes

- `20260917_plan_choice` is a rename plus a single `UPDATE` over `"user"`. On a large table the update is the cost; it is one statement inside the migration's transaction.
- Nothing needs to be run by hand: the backfill is the existing-user rule.
- The screen states the trial length from `BILLING_TRIAL_DAYS`. With it at 0 the copy says "Secure checkout shows the final amount" instead of promising "0 days free".
- The plugin side of this is one MR on the EE repo, which adds `GET`/`POST /billing/plan-choice` and the checkout stamping. The submodule pointer is deliberately not bumped here.
- No Stripe call was made from a developer machine and no email was sent: every provider call in the tests is mocked.

## Not in scope

- Changing what the plans are, what they cost, or what they include. The cards render whatever the published content says.
- The self-hosted ladder: the console sells hosted subscriptions only, and the screen filters to cloud plans exactly as the plan page does.
- Asking an existing account to revisit its plan. The backfill deliberately treats every account that predates this as settled.
- Anything about the trial's own mechanics (card collection, expiry, `missing_payment_method`), which is unchanged.

> [!NOTE]
> **Low risk**
> No security issues found; the schema change is a rename-plus-backfill migration with documented lock cost, every fail-safe default points away from blocking the user, and the behaviour is pinned by 36 new tests.
>
> **Overview**
> Replaces the dismissible post-signup trial dialog with a persisted `plan_choice_made_at` stamp on the user and a non-dismissible first-login plan-choice screen in the cloud console. Pricing-page arrivals record their plan at registration via `plan_choice` on `POST /auth/register`; a migration backfill settles every pre-existing account so nobody is interrupted.
>
> <sup>Written by [Preloop PR Reviewer](https://preloop.ai) for commit 4055f2b5. Updates automatically on new commits.</sup>

### CODE REVIEW

File: backend/preloop/api/auth/router.py [hunk 1]
```diff
@@ -598,6 +605,15 @@ async def register(
             "is_active": True,
             "email_verified": False,
             "user_source": "local",
+            # Somebody who arrived from the pricing page already answered the
+            # plan question, and the console must not ask it again. Recording
+            # it here, on the row, is what carries the answer through the
+            # email verification round trip and onto every other device: a
+            # query string does not survive a link in an inbox, and
+            # localStorage does not survive a second browser.
+            "plan_choice_made_at": (
```
[the remark this answers is not in the document]
Comment (d-mo) [reply backend/preloop/api/auth/router.py:614]: Fixed in 8d241ab, with tests.

You are right that the door was open, and that what comes through it is exactly the state this PR exists to close: `?plan=team` passes the shape check, stamps `plan_choice_made_at`, and the value is then discarded, so the account sits on no plan with the choice suppressed for good. Nothing we mint says that, but "no production path mints it" is not the same as "nobody can".

`register-view.ts` now checks the id against the published price list before sending it. New exported `isFreePlanId(plans, id)` keeps an id only when the plan exists in the loaded cloud ladder and costs nothing on either interval; a plan with no price on either interval is quoted, not free, and is refused too (shape, not plan id, for the same reason the pricing page uses shape). Everything else, a paid id, an invented id, or a content file that will not load, degrades to "ask on first login".

Three details worth calling out:

- The list is `loadPricingContent()`, the same public, cacheable file the pricing page already reads, and it is fetched **only when a plan arrived on the link**. A plain signup makes no extra request; there is a test asserting zero `/landing-content.json` calls for that path.
- A failed load drops the plan rather than trusting it. The cost of dropping is one extra screen on first login; the cost of keeping is the phantom-choice state. The safe direction is to ask.
- The check is held as a promise and awaited in `handleRegister`, so submitting before it lands waits rather than races. "Whether the stamp happens depends on typing speed" is not a rule anyone could debug later.

Core deliberately still knows nothing about the catalog: the server keeps its coarse shape-and-length rule (it must never refuse a signup over a query string), and the narrowing is done where the catalog is already public knowledge.

Five new tests in `register-view.test.ts` (35 passed, was 27): the four `isFreePlanId` cases, the hand-edited paid id, an id the list has never heard of, the unreadable list, and the no-extra-request assertion. Whole frontend suite 2978 passed, 0 failed; `npm run build` and `tsc --noEmit` clean.

File: frontend/src/views/authed/console-shell.ts [hunk 2]
```diff
@@ -467,6 +489,47 @@ export class ConsoleShell extends LitElement {
...
+   *    after this shipped by somebody who did not pick a plan on the way in.
+   *    Everybody else is settled here, for free, on a response the shell had
+   *    already fetched.
+   *
+   * Only then does the billing plugin get the last word, because only it can
+   * see a subscription row or whether this member may buy for the account.
+   */
+  private _startPlanChoiceCheck(profile: UserProfile | null): void {
+    if (this.features['billing'] !== true) return;
+    if (profile?.plan_choice_made !== false) return;
```
[the remark this answers is not in the document]
Comment (d-mo) [reply frontend/src/views/authed/console-shell.ts:516]: Fixed in 0b164b0, with tests.

You are right, and the comment was describing the shape I intended rather than the code I wrote. I took the memo option rather than the comment-only one, because the cohort is real: `not_billing_actor` survives the rbac stamp for anyone created before this ships, and `subscription_exists` covers any account whose subscription arrived by a path that did not stamp the user (a webhook-only reconcile, a support-created subscription). Those people would have paid one request per console load, forever, to be told the same thing.

`getPlanChoice` now remembers a refusal **that carries a reason**, for the tab, and answers subsequent calls from it. The distinction you drew is the exact condition in the code:

```ts
if (!state.show && state.reason !== "unavailable") {
  planChoiceSettled = state;
}
```

- `unavailable` is never remembered. That is what a 404, a 500, a parse failure and a dropped connection all produce, so the question still heals itself when the plugin comes back.
- `show: true` is never remembered either, or the screen could not appear on a later load after a refresh.
- `invalidateApiCaches()` clears it, alongside the features and profile caches. Whose plan question is settled is a fact about one person, and signing out and back in as somebody else must not inherit it.

The `plan_choice_made` doc comment now says "at most one request per page load, only while the account is still being asked" and points at the memo, and the gate comment in `console-shell.ts` says the same thing in the place a reader meets the gates.

Three new tests in `api.test.ts` (67 passed, was 64): an unreachable plugin asks again and a reasoned refusal asks once (same test, so the difference is visible in one place); `show: true` keeps asking; and the memo is dropped by `invalidateApiCaches`. The existing plan-choice block got a `beforeEach(invalidateApiCaches)` so the per-module memo cannot leak between cases. Whole frontend suite 2981 passed, 0 failed; build and `tsc --noEmit` clean.
